Readme.md

```diff
@@ -15,9 +15,50 @@ Supports POSTing artifacs, directory listing, meta tags, locking, and simple tri
 
 <!-- /TOC -->
 
-## Directory listing
+## REST API
+
+### Directory Tree Endpoint
+
+- **Method:** GET
+- **Base URL:** `/api/dirtree`
```

Comment:
Instead of `dirtree` endpoint, I would prefer the use of `Accept` header. 
So if `Accept: application/json` then behave like this new endpoint, otherwise send the original html response. As I see the webui would be able to use it easily too.

Reply:
Requiring a new header is little more complex design than having a base path for the API, and e.g., it prevents directly demonstrating the REST API through a browser. Especially for an artifactory where the selling point is simplicity. :)  For novice users, it is not straightforward to figure out why accessing the same URL results in different output...

Other options to consider:

- if we do not insist backwards compatibility, we can provide the current non-js html output on different base path than '/', .e.g, "/dirtree".
- we can move the API and the browser.html to a different port.

Anyway, do we really need to keep the non-js html output? Is there a use case where that is beneficiary, except backwards compatilibity?